uplink/metainfo/client.go

```diff
@@ -234,3 +234,116 @@ func (client *Client) GetProjectInfo(ctx context.Context) (resp *pb.ProjectInfoR
 
 	return client.client.ProjectInfo(ctx, &pb.ProjectInfoRequest{})
 }
+
+// CreateBucket creates a new bucket
+func (client *Client) CreateBucket(ctx context.Context, bucket storj.Bucket) (_ storj.Bucket, err error) {
+	defer mon.Task()(&ctx)(&err)
+	req := convertBucketToProtoRequest(bucket)
+	resp, err := client.client.CreateBucket(ctx, &req)
+	if err != nil {
+		if status.Code(err) == codes.NotFound {
+			return storj.Bucket{}, storj.ErrBucketNotFound.Wrap(err)
+		}
+		return storj.Bucket{}, Error.Wrap(err)
+	}
+
+	bucket = convertProtoToBucket(resp.Bucket)
+	return bucket, nil
+}
+
+// GetBucket returns a bucket
+func (client *Client) GetBucket(ctx context.Context, bucketName string) (_ storj.Bucket, err error) {
+	defer mon.Task()(&ctx)(&err)
+	resp, err := client.client.GetBucket(ctx, &pb.BucketGetRequest{Name: []byte(bucketName)})
+	if err != nil {
+		if status.Code(err) == codes.NotFound {
+			return storj.Bucket{}, storj.ErrBucketNotFound.Wrap(err)
+		}
+		return storj.Bucket{}, Error.Wrap(err)
+	}
+	bucket := convertProtoToBucket(resp.Bucket)
+	return bucket, nil
+}
+
+// DeleteBucket deletes a bucket
+func (client *Client) DeleteBucket(ctx context.Context, bucketName string) (err error) {
+	defer mon.Task()(&ctx)(&err)
+	_, err = client.client.DeleteBucket(ctx, &pb.BucketDeleteRequest{Name: []byte(bucketName)})
+	if err != nil {
+		if status.Code(err) == codes.NotFound {
+			return storj.ErrBucketNotFound.Wrap(err)
+		}
+		return Error.Wrap(err)
+	}
+	return nil
+}
+
+// ListBuckets lists buckets
+func (client *Client) ListBuckets(ctx context.Context, listOpts storj.BucketListOptions) (_ storj.BucketList, err error) {
+	defer mon.Task()(&ctx)(&err)
+	req := &pb.BucketListRequest{
+		Cursor: []byte(listOpts.Cursor),
+		Limit:  int32(listOpts.Limit),
+	}
+	resp, err := client.client.ListBuckets(ctx, req)
+	if err != nil {
+		return storj.BucketList{}, Error.Wrap(err)
+	}
+	resultBucketList := storj.BucketList{}
+	resultBucketList.Items = make([]storj.Bucket, len(resp.GetItems()))
+	for i, item := range resp.GetItems() {
+		resultBucketList.Items[i] = storj.Bucket{
+			Name:    string(item.GetName()),
+			Created: item.GetCreatedAt(),
+		}
+	}
+	bucketList := storj.BucketList{
```

Comment:
why creating this `bucketList` when we have `resultBucketList`?

Reply:
good question. idk. i think i was tired when writing this code. good catch.